pkg/kubelet/cm/cpumanager/cpu_manager_test.go

```diff
@@ -639,15 +643,8 @@ func TestCPUManagerGenerate(t *testing.T) {
 				if rawMgr.policy.Name() != testCase.expectedPolicy {
 					t.Errorf("Unexpected policy name. Have: %q wants %q", rawMgr.policy.Name(), testCase.expectedPolicy)
 				}
-				if rawMgr.policy.Name() == string(PolicyNone) {
```

Comment:
with my changes this is always false... (hence deleted)

Reply:
I agree with such architectural decision in the code itself. But let's think about meaning of the return value of GetAssignableCPUs, it's  CPUs available for exclusive allocation, that's why static_policy  evaluates it and exclude reserved cpus. So we need to decide, what is assignable cpu in case of none policy, because where is no exclusive isolation in this case. So I agree with removed code, that nil topology for none policy is expected topology.

Reply:
That's a good point, thanks for raising. I see as viable only this approach or report nothing, because if this `GetAssignableCPUs` needs to report cpus available for exclusive allocation, policy None must report empty set (which is by side effect also greatly simplifies the code!). It really depends on clarifying this point and having it specified.

Reply:
By definition, assignable CPUs are the ones that can be exclusively allocated to pods that meet the exclusivity requirement (ie guaranteed QoS class and integral CPU request). I see the reason why previously GetAssignableCPUs was not included in the Policy interface as assignability of CPUs as a concept is only applicable in case of static policy i.e. scenarios where workloads CAN get exclusive access to core(s). By default (in case of None policy), kubernetes relies on CFS quota to enforce CPU limits on pod containers and not on cpu affinity for cpu assignment. 
So, passing back an empty set seems fine to me in case of none policy but it feels like saying that both the polices have assignable CPUs (by adding  GetAssignableCPUs() in the policy interface) and then later in case of none policy returning an empty set but I guess we do that in go all the time :)
Similarly for topology field in case of none policy, first we say that in case of none policy we have a topology and then  always return nil. I wonder what is the motivation for this change? I think if we would like to proceed with these changes, we should add comments in case of none policy too (like we have for static policy [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/policy_static.go#L35)) to capture these for future developers.

Reply:
so from my point of view:
extending policy interface with GetAssignableCPUs - is ok
keeping topology in NonePolicy is unnecessary, since GetAssignableCPUs will always return nil.

Reply:
Works for me and as nice bonus makes the code simpler :)
I will summarize this conversation in a code comment.